src/ephemeral.rs

```diff
@@ -588,4 +593,51 @@ mod tests {
 
         Ok(())
     }
+
+    #[async_std::test]
+    async fn test_ephemeral_delete_msgs() {
+        let t = TestContext::new_alice().await;
+        let chat = t.get_self_chat().await;
+
+        t.send_text(chat.id, "Saved message, which we delete manually")
+            .await;
+        let msg = t.get_last_msg(chat.id).await;
+        msg.id.delete_from_db(&t).await.unwrap();
+        check_msg_was_deleted(&t, &chat, msg.id).await;
+
+        chat.id
+            .set_ephemeral_timer(&t, Timer::Enabled { duration: 1 })
+            .await
+            .unwrap();
+        let msg = t
+            .send_text(chat.id, "Saved message, disappearing after 1s")
+            .await;
+
+        sleep(Duration::from_millis(1100)).await;
```

Comment:
huh, i guess i didn't realise this would need sleep.  That's not ideal for the rust unittests I think, but fine for this I guess.  I wonder if we can in the future figure out a way to control time better during tests as well, I believe Tokio has something for this, not sure about async-std

Reply:
As cargo runs tests in parallel, it's not as bad as it sounds in the first moment but of course it's not nice, either.